base/setup/lib/bootsup.c

```diff
@@ -1228,6 +1228,107 @@ InstallBtrfsBootcodeToPartition(
     return STATUS_SUCCESS;
 }
 
+static
+NTSTATUS
+InstallNtfsBootcodeToPartition(
+    IN PUNICODE_STRING SystemRootPath,
+    IN PUNICODE_STRING SourceRootPath,
+    IN PUNICODE_STRING DestinationArcPath)
+{
+    NTSTATUS Status;
+    BOOLEAN DoesFreeLdrExist;
+    WCHAR SrcPath[MAX_PATH];
+    WCHAR DstPath[MAX_PATH];
+
+    /* NTFS partition */
+    DPRINT("System path: '%wZ'\n", SystemRootPath);
+
+    /* Copy FreeLoader to the system partition, always overwriting the older version */
+    CombinePaths(SrcPath, ARRAYSIZE(SrcPath), 2, SourceRootPath->Buffer, L"\\loader\\freeldr.sys");
+    CombinePaths(DstPath, ARRAYSIZE(DstPath), 2, SystemRootPath->Buffer, L"freeldr.sys");
+
+    DPRINT1("Copy: %S ==> %S\n", SrcPath, DstPath);
+    Status = SetupCopyFile(SrcPath, DstPath, FALSE);
+    if (!NT_SUCCESS(Status))
+    {
+        DPRINT1("SetupCopyFile() failed (Status %lx)\n", Status);
+        return Status;
+    }
+
+    /* Prepare for possibly updating 'freeldr.ini' */
+    DoesFreeLdrExist = DoesFileExist_2(SystemRootPath->Buffer, L"freeldr.ini");
+    if (DoesFreeLdrExist)
+    {
+        /* Update existing 'freeldr.ini' */
+        DPRINT1("Update existing 'freeldr.ini'\n");
+        Status = UpdateFreeLoaderIni(SystemRootPath->Buffer, DestinationArcPath->Buffer);
+        if (!NT_SUCCESS(Status))
+        {
+            DPRINT1("UpdateFreeLoaderIni() failed (Status %lx)\n", Status);
+            return Status;
+        }
+
+        return STATUS_SUCCESS;
+    }
+
+    /* Check for *nix bootloaders */
+
+    DPRINT1("Create new 'freeldr.ini'\n");
+
+    /* Certainly SysLinux, GRUB, LILO... or an unknown boot loader */
+    DPRINT1("*nix or unknown boot loader found\n");
+
+    if (IsThereAValidBootSector(SystemRootPath->Buffer))
+    {
+        PCWSTR BootSector = L"BOOTSECT.OLD";
+
+        Status = CreateFreeLoaderIniForReactOSAndBootSector(
+                     SystemRootPath->Buffer, DestinationArcPath->Buffer,
```

Comment:
for NTFS partition installation it would make more sense to check for Windows NT+ bootloaders instead (see the other existing code where ntldr is checked for existence)

Reply:
Why?